http4k-aws/src/main/kotlin/org/http4k/aws/AwsPreSignRequests.kt

```diff
@@ -0,0 +1,39 @@
+package org.http4k.aws
+
+import org.http4k.core.Request
+import org.http4k.filter.Payload
+import java.time.Clock
+import java.time.Duration
+import java.util.Locale
+
+class AwsPreSignRequests(
+    private val scope: AwsCredentialScope,
+    private val credentials: AwsCredentials,
+    private val clock: Clock = Clock.systemDefaultZone()
+) {
+    operator fun invoke(request: Request, expires: Duration): AwsPreSignedRequest {
+        val time = clock.instant()
+        val awsDate = AwsRequestDate.of(time)
+
+        val headers = request.headers + ("host" to request.uri.host)
+        val signedHeaders = headers.map { it.first.lowercase(Locale.getDefault()) }.toSet().sorted().joinToString(";")
```

Comment:
I could get this from the `AwsCanonicalRequest`, but I need these to make the `AwsCanonicalRequest`.  🤷 

Worth extracting the common code?

Reply:
Yes - let's try to isolate it, even if it's an internal scope :)

Reply:
Done